iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/fileio/FileIOManager.java

```diff
@@ -33,9 +33,15 @@ public class FileIOManager {
   @Qualifier("LocalFileIO")
   FileIO localFileIO;
 
+  @Autowired(required = false) // doesn't inject if null
+  @Qualifier("ADLSFileIO") // avoid ambiguity with multiple beans of the same type
+  FileIO adlsFileIO;
```

Comment:
Let it autowire `ADLSFileIO` directly instead of `qualifier + FileIO`.

Reason why we used qualifier before: Because HDFS and Local use same HdfsFileIO